eq-author/src/components/PositionModal/index.js

```diff
@@ -1,149 +1,118 @@
-
-    this.state = {
-      isOpen: false,
-      selectedPosition: props.selected.position,
-      previousSelectedPosition: props.selected.position,
-    };
-  }
-
-  handleClose = () => {
-    const { previousSelectedPosition } = this.state;
-
-    this.setState({
-      isOpen: false,
-      selectedPosition: previousSelectedPosition,
-    });
-  };
-
-  handleOpen = () => {
-    this.setState({
-      isOpen: true,
-    });
-  };
-
-  handleChange = ({ value }) => {
-    this.setState({ selectedPosition: parseInt(value, 10) });
-  };
-
-  handleConfirm = e => {
-    const { onMove } = this.props;
-    const { selectedPosition } = this.state;
-    e.preventDefault();
-    this.setState({ isOpen: false }, () => onMove(selectedPosition));
-  };
-
-  getOrderedOptions() {
-    const { selected, options } = this.props;
-    const { selectedPosition } = this.state;
-
-    return moveTo(options, selected, selectedPosition);
-  }
-
-  renderPositionSelect(data) {
-    const { isOpen, selectedPosition } = this.state;
-    return (
-      <ItemSelectModal
-        data-test={"position-select-modal"}
-        title={"Position"}
-        primaryText={"Move"}
-        isOpen={isOpen}
-        onClose={this.handleClose}
-        onConfirm={this.handleConfirm}
-      >
-        <ItemSelect
-          data-test={"position-item-select"}
-          name={"position"}
-          value={String(selectedPosition)}
-          onChange={this.handleChange}
-        >
-          {data.map((item, i) => (
-            <Option key={i} value={String(i)}>
-              {item.displayName}
-            </Option>
-          ))}
-        </ItemSelect>
-      </ItemSelectModal>
-    );
-  }
-
-  render() {
-    const options = this.getOrderedOptions();
-    const positionButtonId = uniqueId("PositionModal");
-    return (
-      <div data-test={"position-modal"}>
-        <Label htmlFor={positionButtonId}>Position</Label>
-        <Trigger
-          data-test="position-modal-trigger"
-          id={positionButtonId}
-          onClick={this.handleOpen}
-        >
-          Select
-        </Trigger>
-        {this.renderPositionSelect(options)}
-      </div>
-    );
-  }
-}
-
-export default PositionModal;
+import React, { useState, useMemo } from "react";
+import PropTypes from "prop-types";
+import ItemSelectModal from "components/ItemSelectModal";
+import ItemSelect, { Option } from "components/ItemSelectModal/ItemSelect";
+import { uniqueId } from "lodash";
+import Icon from "assets/icon-select.svg";
+import styled from "styled-components";
+
+import { colors, radius } from "constants/theme";
+
+const Label = styled.label`
+  display: block;
+  font-size: 1em;
+  font-weight: bold;
+  margin-bottom: 0.25rem;
+  margin-top: 1.25rem;
+`;
+
+const Trigger = styled.button.attrs({ type: "button" })`
+  width: 100%;
+  font-size: 1em;
+  padding: 0.5rem 2em 0.5rem 0.5rem;
+  background: white url('${Icon}') no-repeat right center;
+  border: solid 1px #aeaeae;

```

Comment:
Whilst we're in this file, we could swap this colour out for one that's in our Theme library.

Reply:
Sure thing